rotkehlchen/chain/evm/decoding/aave/v3/decoder.py

```diff
@@ -180,12 +186,47 @@ def _decode_interest(
                     event.notes = f'Receive {event.amount} {event.asset.symbol_or_name()} from migrating an AAVE v2 position to v3'  # noqa: E501
                     event.address = None  # no need to have zero address
                     migratein_event = event
+                elif swap_event is not None and swap_event.extra_data is not None:  # logic for collateral swap  # noqa: E501
+                    if (
+                        received_addr := swap_event.extra_data.get('received_addr')
+                    ) is None:
+                        log.error(f'Aave V3 swap event {transaction} missing extradata. Skipping')
+                        continue
+
+                    if (
+                        received_token := event.asset.resolve_to_evm_token()
+                    ).underlying_tokens is None:
+                        log.error(f'Aave V3 asset {received_token.identifier} expected to have underlying tokens. Skipping')  # noqa: E501
+                        continue
+
+                    if not any(received_addr == x.address for x in received_token.underlying_tokens):  # noqa: E501
```

Comment:
```suggestion
                    if not any(received_addr == x.address for x in (received_token.underlying_tokens or [])):  # noqa: E501
```
I think this would allow you to remove the `if underlying_tokens is None` check above.